crypto/evp/evp_lib.c

```diff
@@ -688,19 +688,7 @@ const OSSL_PROVIDER *EVP_MD_provider(const EVP_MD *md)
 
 int EVP_MD_block_size(const EVP_MD *md)
 {
-    int ok;
-    size_t v;
-    OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
-
-    if (md == NULL) {
-        ERR_raise(ERR_LIB_EVP, EVP_R_MESSAGE_DIGEST_IS_NULL);
-        return -1;
-    }
```

Comment:
Why do we don't check for md here but check in EVP_MD_size?

Reply:
The same question is about EVP_MD_flags

Reply:
I'm just reverting the implementation back to the way it used to be in 1.1.1. There, EVP_MD_size() has the check:

https://github.com/openssl/openssl/blob/64a1b940d2b640e5edf0feae90e81bbb6b4941e7/crypto/evp/evp_lib.c#L320-L327

But EVP_MD_block_size and EVP_MD_flags do not:
https://github.com/openssl/openssl/blob/64a1b940d2b640e5edf0feae90e81bbb6b4941e7/crypto/evp/evp_lib.c#L305-L308
https://github.com/openssl/openssl/blob/64a1b940d2b640e5edf0feae90e81bbb6b4941e7/crypto/evp/evp_lib.c#L329-L332